lib/LoginStrategy.ts

```diff
@@ -351,6 +425,27 @@ export class LoginStrategy extends BaseStrategy {
     );
   }
 
+  /**
+   * extracts siwe message and signature either from request body or query
+   * @param req
+   * @returns {string} siwe object - message and signature
+   */
+  extractSiwe(req: Request): SiweMessageReqPayload {
+    if (req.body.identity)
+      return {
+        signature:
+          lookup(req.body.identity, 'signature') ||
+          lookup(req.query, 'signature'),
+        message: req.body.message,
+      };
+
+    return {
+      signature:
+        lookup(req.body, 'signature') || lookup(req.query, 'signature'),
+      message: req.body.message,
+    };
+  }
```

Comment:
It's related to @JGiter 's comment about `identity`, but I think this implementation is a bit overcomplicated.
To me, it makes sense to have the properties on `SiweMessageReqPayload` be mandatory but return `null` if it can't be extracted (this is inline with how `extractToken` is implemented currently).

I also don't really see the value of `lookup`, but maybe I'm missing something there.

```suggestion
  extractSiwe(req: Request): SiweMessageReqPayload || null {
    if (req.body.message && req.body.signature) {
      return {
        signature: req.body.signature,
        message: req.body.message
      };
    }
    else {
     return null;
    }
  }
```